src/frontend/src/tables/settings/GroupTable.tsx

```diff
@@ -205,34 +410,40 @@ export function GroupTable({
     return actions;
   }, [user]);
 
+  const clickable: boolean = useMemo(
+    () => user.hasChangePermission(ModelType.group) && !directLink,
+    [user, directLink]
+  );
+
   return (
     <>
       {newGroup.modal}
       {deleteGroup.modal}
-      <DetailDrawer
-        title={t`Edit group`}
-        renderContent={(id) => {
-          if (!id || !id.startsWith('group-')) return false;
-          return (
-            <GroupDrawer
-              id={id.replace('group-', '')}
-              refreshTable={table.refreshTable}
-            />
-          );
-        }}
-      />
+      {user.hasViewRole(UserRoles.admin) && (
+        <DetailDrawer
+          size='xl'
+          title={t`Edit Group`}
+          renderContent={(id) => {
+            if (!id || !id.startsWith('group-')) return false;
+            return (
+              <GroupDrawer
+                id={id.replace('group-', '')}
+                refreshTable={table.refreshTable}
+              />
+            );
+          }}
+        />
+      )}
       <InvenTreeTable
         url={apiUrl(ApiEndpoints.group_list)}
         tableState={table}
         columns={columns}
         props={{
-          rowActions: directLink ? undefined : rowActions,
+          rowActions: clickable ? rowActions : undefined,
           tableActions: tableActions,
-          onRowClick: directLink
-            ? undefined
-            : (record) => openDetailDrawer(record.pk),
-
-          modelType: ModelType.group
+          onRowClick: clickable
+            ? (record) => openDetailDrawer(record.pk)
+            : undefined
```

Comment:
So we loose the detail page link?

Reply:
The change here is to make the `onRowClick` and `rowActions` attributes return `undefined` if there are no actions available - this ensures that the mouse pointer is correct.

Reply:
I have now reverted this include the old behavior too